### PR TITLE
added feature/transfer/withdrawal-page

### PR DESCRIPTION
**Pull request by gabito1451**
Added transfer and withdrawal page

### CODE REVIEW

File: transfer.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+const transferForm = document.getElementById("transfer-form");
+const selectBeneficiaryAccountNumberElem = document.getElementById(
+  "beneficiary-account-number"
+);
+const transferAmountElem = document.getElementById("transfer-amount");
+const accountPinElem = document.getElementById("account-pin");
+const currentUser = getUserByAccountNumber(currentUserAccountNumber);
+const allUsers = getAllUsers();
+// populate users in our database into the select field and remove the current user account number
+allUsers.forEach((user) => {
```
Comment (nwankwo-ikemefuna) [inline transfer.js]: Suggested:

```
allUsers.
    .filter((user) => user.accountNumber !== currentUserAccountNumber)
    .forEach((user) => {
      const optionElem = document.createElement("option");
      optionElem.value = user.accountNumber;
      optionElem.textContent = `${user.accountNumber}`;
      selectBeneficiaryAccountNumberElem.append(optionElem);
    });
```

`allUsers.filter` will filter out the current user object from the `allUsers` array, leaving you with the rest of the users to apply `.forEach` on.

File: transfer.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  if (accountPinElem.value !== currentUser.accountPin) {
+    alert("Incorrect PIN.");
+    return;
+  }
+  if (transferAmountElem.value > currentBalance) {
+    alert("Insufficient Balance");
+    return;
+  }
+  if (transferAmountElem.value < 0) {
```
Comment (nwankwo-ikemefuna) [inline transfer.js]: This check should be `(transferAmountElem.value < 1)`. You can't send 0 Naira to someone.

File: withdraw.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  if (accountPinElem.value !== currentUser.accountPin) {
+    alert("Incorrect PIN.");
+    return;
+  }
+  if (withdrawalAmountElem.value > currentBalance) {
+    alert("Insufficient Balance");
+    return;
+  }
+  if (withdrawalAmountElem.value < 0) {
```
Comment (nwankwo-ikemefuna) [inline withdraw.js]: This check should be (withdrawalAmountElem.value < 1). You can't withdraw 0 Naira.

**Review by nwankwo-ikemefuna — CHANGES_REQUESTED** [review]
Looks good. Just few changes.

File: withdraw.js (hunk 3 above)
Comment (gabito1451) [reply withdraw.js]: Done

File: transfer.js (hunk 2 above)
Comment (gabito1451) [reply transfer.js]: Done

File: transfer.js (hunk 1 above)
Comment (gabito1451) [reply transfer.js]: Done

File: transfer.js [hunk 4]
```diff
@@ -7,20 +7,14 @@ const accountPinElem = document.getElementById("account-pin");
...
-  optionElem.value = user.accountNumber;
-  optionElem.textContent = `${user.accountNumber}`;
-  selectBeneficiaryAccountNumberElem.append(optionElem);
-});
+allUsers
+  .filter((user) => user.accountNumber !== currentUserAccountNumber)
+  .forEach((user) => {
+    const optionElem = document.createElement("option");
+    optionElem.value = user.accountNumber;
+    optionElem.textContent = `${user.accountNumber}`;
```
Comment (nwankwo-ikemefuna) [inline transfer.js]: The option text should be the user's name, not account number.

Screenshot at Feb 16 12-58-45 PM

Change to:
`optionElem.textContent = user.accountName;`

**Review by nwankwo-ikemefuna — CHANGES_REQUESTED** [review]
The balance section should be in all session pages.
Screenshot at Feb 16 1-03-21 PM

On the transactions page, change "Beneficiary account number" to "Beneficiary" and display the beneficiary's name instead of the account number.
Screenshot at Feb 16 1-02-47 PM

File: transfer.js (hunk 4 above)
Comment (gabito1451) [reply transfer.js]: Done

**Review by nwankwo-ikemefuna — APPROVED** [review]

**Pull request merged**